xdist/remote.py

```diff
@@ -123,6 +123,18 @@ def pytest_logwarning(self, message, code, nodeid, fslocation):
             fslocation=str(fslocation),
         )
 
+    # the pytest_warning_captured hook was introduced in pytest 3.8
+    if hasattr(_pytest.hookspec, "pytest_warning_captured"):
+
+        def pytest_warning_captured(self, warning_message, when, item):
+            self.sendevent(
+                "warning_captured",
+                warning_message_data=serialize_warning_message(warning_message),
+                when=when,
+                # XXX what to do with "item" parameter? It can't be serialized
```

Comment:
😬 

In retrospect the `item` parameter was a mistake, it should have been `nodeid`. 😓